src/PrivateDns/PrivateDns/Zones/SetAzurePrivateDnsZone.cs

```diff
@@ -0,0 +1,104 @@
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = FieldsParameterSetName)]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "A hash table which represents resource tags.", ParameterSetName = ResourceParameterSetName)]
+        public Hashtable Tags { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The zone object to set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PrivateDnsZone Zone { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ObjectParameterSetName)]
+        [Parameter(Mandatory = false, HelpMessage = "Do not use the ETag field of the RecordSet parameter for optimistic concurrency checks.", ParameterSetName = ResourceParameterSetName)]
+        public SwitchParameter Overwrite { get; set; }
+
+        [Parameter( ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipeline = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+
+            PrivateDnsZone result = null;
+            PrivateDnsZone zoneToUpdate = null;
+
+            if (this.ParameterSetName == FieldsParameterSetName || this.ParameterSetName == ResourceParameterSetName)
+            {
+                if (!string.IsNullOrEmpty(this.ResourceId))
+                {
+                    PrivateDnsUtils.GetResourceGroupNameAndZoneNameFromResourceId(this.ResourceId, out var resourceGroupName, out var zoneName);
+                    this.ResourceGroupName = resourceGroupName;
+                    this.Name = zoneName;
+                }
+
+                if (this.Name.EndsWith("."))
+                {
+                    this.Name = this.Name.TrimEnd('.');
+                    this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{this.Name}\".");
+                }
+
+                zoneToUpdate = this.PrivateDnsClient.GetPrivateDnsZone(this.Name, this.ResourceGroupName);
+                zoneToUpdate.Etag = "*";
+                zoneToUpdate.Tags = this.Tags;
+            }
+            else if (this.ParameterSetName == ObjectParameterSetName)
+            {
+                if ((string.IsNullOrWhiteSpace(this.Zone.Etag) || this.Zone.Etag == "*") && !this.Overwrite.IsPresent)
+                {
+                    throw new PSArgumentException(string.Format(ProjectResources.Error_EtagNotSpecified, typeof(PrivateDnsZone).Name));
+                }
+
+                zoneToUpdate = this.Zone;
+            }
+
+            if (zoneToUpdate?.Name != null && zoneToUpdate.Name.EndsWith("."))
```

Comment:
If zoneToUpdate is null, should not we be throwing an error the zone does not exist?

Reply:
its a mandatory parameter. I believe passing no zone param would throw an error to the user. This null check can be removed, had added it based on visual studio suggestion.

Reply:
Added a throw.